### PR TITLE
feat(elements|ino-snackbar): add property to make snackbar stay visible on hover

### LINKED ISSUE #330: ino-snackbar: On hover snack bar should not disappear
A request came in for the snack bar not to disappear when hovering over a snack bar.

**Acceptance criteria**
* There is a new boolean property `stayVisibleOnHover` (or similar) that prevents the snackbar from disappearing when hovering over it.
* Timeout will be stopped when hovering over it 
* Timeout continues when user stops hovering over the snackbar

PS: Please also update ELEMENTS-702 in Jira when start/end working on this issue.

### PR DESCRIPTION
**Pull request by AlessaRad**
Closes #330

### CODE REVIEW

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx [hunk 1]
```diff
@@ -67,6 +73,14 @@ export class Snackbar implements ComponentInterface {
       this.handleSnackbarHide(e)
     );
     this.configureTimeout();
+    if (this.inoStayVisibleOnHover) {
+      this.snackbarElement.addEventListener('mouseenter', () => {
+        if (this.timeout) clearTimeout(this.timeout);
```
Comment (janivo) [inline packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:78]: We can shorten this if we extract this to its own function. e.g.:
```ts
componentDidLoad() {
	// ...
	this.snackbarElement.addEventListener('mouseenter', this.interruptTimeout);
	// ...
}

disconnectedCallback() {
	// ...
	this.snackbarElement.removeEventListener('mouseenter', interruptTimeout);
	// ...
}

interruptTimeout(): void {
	if (this.timeout) clearTimeout(this.timeout);
}
```

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx [hunk 2]
```diff
@@ -67,6 +73,14 @@ export class Snackbar implements ComponentInterface {
...
+      });
+      this.snackbarElement.addEventListener('mouseleave', () => {
+        this.configureTimeout();
+      });
```
Comment (janivo) [inline packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:82]: Can be shortened as well:
```suggestion
      this.snackbarElement.addEventListener('mouseleave', this.configureTimeout);
```

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx [hunk 3]
```diff
@@ -75,12 +89,21 @@ export class Snackbar implements ComponentInterface {
     this.snackbarElement.removeEventListener('MDCSnackbar:closing', (e) =>
       this.handleSnackbarHide(e)
     );
+    this.snackbarElement.removeEventListener('mouseenter', () => {
+      if (this.timeout) clearTimeout(this.timeout);
+    });
+    this.snackbarElement.removeEventListener('mouseleave', () => {
+      this.configureTimeout();
+    });
```
Comment (janivo) [inline packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:97]:
```suggestion
    this.snackbarElement.removeEventListener('mouseenter', this.interruptTimeout);
    this.snackbarElement.removeEventListener('mouseleave', this.configureTimeout);
```

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx [hunk 4]
```diff
@@ -75,12 +89,21 @@ export class Snackbar implements ComponentInterface {
...
   }
 
   private configureTimeout() {
     this.snackbarInstance.timeoutMs = -1;
     if (this.inoTimeout >= 0) {
-      setTimeout(() => this.snackbarInstance.close(), this.inoTimeout);
+      this.timeout = setTimeout(
+        () => this.snackbarInstance.close(),
+        this.inoTimeout
+      );
```
Comment (janivo) [inline packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:106]:
```suggestion
      this.timeout = setTimeout(this.snackbarInstance.close, this.inoTimeout);
```

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx [hunk 5]
```diff
@@ -75,12 +89,21 @@ export class Snackbar implements ComponentInterface {
     this.snackbarElement.removeEventListener('MDCSnackbar:closing', (e) =>
       this.handleSnackbarHide(e)
     );
```
Comment (janivo) [inline packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:91]:
```suggestion
    this.snackbarElement.removeEventListener('MDCSnackbar:closing', this.handleSnackbarHide);
```

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx [hunk 6]
```diff
@@ -67,6 +73,14 @@ export class Snackbar implements ComponentInterface {
       this.handleSnackbarHide(e)
```
Comment (janivo) [inline packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:73]:
```ts
this.snackbarElement.addEventListener('MDCSnackbar:closing', this.handleSnackbarHide)
```

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx [hunk 7]
```diff
@@ -75,12 +89,21 @@ export class Snackbar implements ComponentInterface {
...
     );
+    this.snackbarElement.removeEventListener('mouseenter', () => {
+      if (this.timeout) clearTimeout(this.timeout);
+    });
+    this.snackbarElement.removeEventListener('mouseleave', () => {
+      this.configureTimeout();
+    });
   }
 
   private configureTimeout() {
```
Comment (janivo) [inline packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:100]: Maybe we should call it something like `setupTimeout` to make its functionality more clear? What do you think?

**Review by janivo — CHANGES_REQUESTED** [review]
Hey @AlessaRad,
great work! There are some small refactorings I would suggest. What do you think about it?

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx (hunk 1 above)
Comment (AlessaRad) [reply packages/elements/src/components/ino-snackbar/ino-snackbar.tsx]: Good suggestion! I applied it

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx (hunk 3 above)
Comment (AlessaRad) [reply packages/elements/src/components/ino-snackbar/ino-snackbar.tsx]: Applied

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx (hunk 4 above)
Comment (AlessaRad) [reply packages/elements/src/components/ino-snackbar/ino-snackbar.tsx:106]: I think this simplification ist not possible. The MDC close() method is not implemented as an arrow function, hence there would be an internal error.

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx (hunk 7 above)
Comment (AlessaRad) [reply packages/elements/src/components/ino-snackbar/ino-snackbar.tsx]: Seems reasonable, I changed it.

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx (hunk 6 above)
Comment (AlessaRad) [reply packages/elements/src/components/ino-snackbar/ino-snackbar.tsx]: Applied

File: packages/elements/src/components/ino-snackbar/ino-snackbar.tsx (hunk 2 above)
Comment (AlessaRad) [reply packages/elements/src/components/ino-snackbar/ino-snackbar.tsx]: Applied

**Review by janivo — APPROVED** [review]

**Pull request merged**